### PR TITLE
UAR-134: Update journey add individual BOs details page

### PR DESCRIPTION
**Pull request by sjahir**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-134

### Change description
Add individual BOs details page for the update journey

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/update.beneficial.owner.individual.controller.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  DateOfBirthKey,
+  DateOfBirthKeys,
+  StartDateKey,
+  StartDateKeys,
+} from "../../model/date.model";
+import { UPDATE_BENEFICIAL_OWNER_INDIVIDUAL_PAGE } from "../../config";
+import { v4 as uuidv4 } from "uuid";
+
+export const get = (req: Request, res: Response) => {
+  logger.debugRequest(req, `GET ${UPDATE_BENEFICIAL_OWNER_INDIVIDUAL_PAGE}`);
```
Comment (AndreKramer7654) [inline src/controllers/update/update.beneficial.owner.individual.controller.ts:45]: logger.debugRequest(req, `${req.method} ${req.route.path}`);

File: src/controllers/update/update.beneficial.owner.individual.controller.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
+import { NextFunction, Request, Response } from "express";
```
[the remark this answers is not in the document]
Comment (sjahir) [reply src/controllers/update/update.beneficial.owner.individual.controller.ts:1]: Hey @AndreKramer7654, ignore the PR for now, still working on it and haven't pushed up any of the latest changes yet

File: src/controllers/beneficial.owner.individual.controller.ts [hunk 3]
```diff
@@ -1,160 +1,25 @@
...
-import {
-  DateOfBirthKey,
-  DateOfBirthKeys,
-  StartDateKey,
-  StartDateKeys,
-} from "../model/date.model";
-import { BENEFICIAL_OWNER_INDIVIDUAL_PAGE, BENEFICIAL_OWNER_TYPE_URL } from "../config";
-import { v4 as uuidv4 } from "uuid";
 
+import { getBeneficialOwnerIndividual, getBeneficialOwnerIndividualById, postBeneficialOwnerIndividual, removeBeneficialOwnerIndividual, updateBeneficialOwnerIndividual } from "../utils/beneficial.owner.individual";
```
Comment (AndreKramer7654) [inline src/controllers/beneficial.owner.individual.controller.ts:3]: maybe have one imported name per line.

File: src/controllers/update/update.beneficial.owner.individual.controller.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import { getBeneficialOwnerIndividual, getBeneficialOwnerIndividualById, postBeneficialOwnerIndividual, removeBeneficialOwnerIndividual, updateBeneficialOwnerIndividual } from "../../utils/beneficial.owner.individual";
```
Comment (AndreKramer7654) [inline src/controllers/update/update.beneficial.owner.individual.controller.ts:3]: maybe have one imported name per line

File: src/controllers/update/update.beneficial.owner.individual.controller.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+export const getById = (req: Request, res: Response, next: NextFunction) => {
+  getBeneficialOwnerIndividualById(req, res, next, UPDATE_BENEFICIAL_OWNER_INDIVIDUAL_PAGE, UPDATE_BENEFICIAL_OWNER_TYPE_URL);
+};
+
+export const post = (req: Request, res: Response, next: NextFunction) => {
+  postBeneficialOwnerIndividual(req, res, next, UPDATE_BENEFICIAL_OWNER_TYPE_URL, false);
+};
+
+export const update = (req: Request, res: Response, next: NextFunction) => {
+  console.log();
```
Comment (AndreKramer7654) [inline src/controllers/update/update.beneficial.owner.individual.controller.ts:20]: Remove

File: src/controllers/update/update.beneficial.owner.individual.controller.ts (hunk 4 above)
Comment (sjahir) [reply src/controllers/update/update.beneficial.owner.individual.controller.ts:3]: yes sir

File: src/controllers/beneficial.owner.individual.controller.ts (hunk 3 above)
Comment (sjahir) [reply src/controllers/beneficial.owner.individual.controller.ts:3]: good idea, was considering this originally

File: views/beneficial-owner-individual.html [hunk 6]
```diff
@@ -1,8 +1,8 @@
 {% extends "layout.html" %}
 
+{% set title = "Tell us about the individual person" %}
```
Comment (AndreKramer7654) [inline views/beneficial-owner-individual.html:3]: Don't see a reason to move this out of the block below

**Review by AndreKramer7654 — APPROVED** [review]

File: views/beneficial-owner-individual.html (hunk 6 above)
Comment (sjahir) [reply views/beneficial-owner-individual.html:3]: I think thats why they don't do that in the other files as well

**Review by bantao-ch — APPROVED** [review]

File: views/beneficial-owner-individual.html [hunk 7]
```diff
@@ -11,65 +11,8 @@
...
-        {% set nameToBeRemoved = first_name + " " + last_name %}
-        {% include "includes/remove-button.html" %}
-
-      </form>
-    </div>
-  </div>
+  {% include "includes/page-templates/beneficial-owner-individual.html" %}
+  {% set removePathURL = OE_CONFIGS.BENEFICIAL_OWNER_INDIVIDUAL_URL + OE_CONFIGS.REMOVE + "/" + id %}
+  {% set nameToBeRemoved = first_name + " " + last_name %}
+  {% include "includes/remove-button.html" %}
```
Comment (markpit) [inline views/beneficial-owner-individual.html:17]: the includes/page-templates/beneficial-owner-individual.html has a remove button included at the bottom of it, but this page (views/beneficial-owner-individual.html) also has a remove button.
looking at the original version of views/beneficial-owner-individual.html, i can only see the one remove button. Does this cause 2 remove buttons to appear in the registration journey?

Comment (markpit) [reply views/beneficial-owner-individual.html:17]: I've checked out the branch and it does show 2 remove buttons when changing an BO individual
Screenshot 2023-02-17 at 15 49 43

Comment (markpit) [reply views/beneficial-owner-individual.html]: Looks ok now, thanks

**Review by markpit — APPROVED** [review]

**Pull request merged**